### PR TITLE
added summarize

### PR DESCRIPTION
**Pull request by ylyangtw**

### CODE REVIEW

**Comment by ylyangtw** [comment]
image

`summarize` would fail if the source exists in the summary namespace.
For testing, I clear the summary namespace first using this sqarql:
```
DELETE ?s ?p ?o  WHERE { ?s ?p ?o .FILTER regex(str(?s), "iris") .}
```

**Comment by valentinedwv** [comment]
In manage blaze graph, there are delete and create namespace, was setup for integration testing, but think it should work.

https://earthcube.github.io/earthcube_utilities/earthcube_utilities/earthcube_utilities_code/#ec.graph.manageGraph.ManageBlazegraph

File: dagster/implnets/templates/v1/implnet_ops_SOURCEVAL.py [hunk 1]
```diff
@@ -697,6 +703,40 @@ def SOURCEVAL_bucket_urls(context):
...
+        summarydf = get_summary4repoSubset(endpoint, source_name)
+
+        nt, g = summaryDF2ttl(summarydf, source_name)  # let's try the new generator
+
+        summaryttl = g.serialize(format='longturtle')
+
+        inserted = sumnsgraph.insert(bytes(summaryttl, 'utf-8'), content_type="application/x-turtle")
+
+        # TO DO: upload to minio
+        if not inserted:
```
Comment (valentinedwv) [inline dagster/implnets/templates/v1/implnet_ops_SOURCEVAL.py:725]: Don't need this.  do need to raise an exception so the step is flagged as bad.

**Review by valentinedwv — CHANGES_REQUESTED** [review]

**Comment by valentinedwv** [comment]
If there is an issue with files loading to the graph,
Then We should probably look to save the file to s3... then load to the graph.

We could better test the loading issues. aka, how can we test this, and catch this, and where should it be done (dagster/ec_utils)
 
Then maybe the saved becomes an actual asset, at some point

> image
> 
> `summarize` would fail if the source exists in the summary namespace. For testing, I clear the summary namespace first using this sqarql:
> 
> ```
> DELETE ?s ?p ?o  WHERE { ?s ?p ?o .FILTER regex(str(?s), "iris") .}
> ```

**Comment by valentinedwv** [comment]
This issue as how to push a file in s3 to a graph.

https://github.com/earthcube/scheduler/issues/64

File: dagster/implnets/templates/v1/implnet_ops_SOURCEVAL.py (hunk 1 above)
Comment (ylyangtw) [reply dagster/implnets/templates/v1/implnet_ops_SOURCEVAL.py:724]: I just removed it and simply raise an exception

**Comment by ylyangtw** [comment]
> If there is an issue with files loading to the graph, Then We should probably look to save the file to s3... then load to the graph.
> 
> We could better test the loading issues. aka, how can we test this, and catch this, and where should it be done (dagster/ec_utils)
> 
> Then maybe the saved becomes an actual asset, at some point
> 
> > image
> > `summarize` would fail if the source exists in the summary namespace. For testing, I clear the summary namespace first using this sqarql:
> > ```
> > DELETE ?s ?p ?o  WHERE { ?s ?p ?o .FILTER regex(str(?s), "iris") .}
> > ```

Agree!

**Comment by ylyangtw** [comment]
image
if loading to graph failed, it will upload the result to s3.

**Comment by valentinedwv** [comment]
pull and check this. I reworked the code

**Comment by valentinedwv** [comment]
Might need to update a dependency to: https://github.com/earthcube/scheduler/blob/51652704c1a7022c1b61c6a09fe8240c15380b65/dagster/implnets/requirements_code.txt#L18

earthcube-utilities @ git+https://github.com/earthcube/earthcube_utilities@b671efb#subdirectory=earthcube_utilities

Should have changed it in the standard requirements.txt, too.

**Comment by valentinedwv** [comment]
Try again. Got the triplestore swapped.
fixed an upload issue.

**Comment by ylyangtw** [comment]
Nice! All work

**Pull request merged**